### PR TITLE
UX: Multichain: Ensure network picker renders properly for long and short names

### PR DESCRIPTION
**Pull request by darkwing**
## Explanation

My previous PR to enforce a maximum network name with ellipsis worked well for long network names but not for short.  This was the side effect with short network names:

SCR-20230502-mduz

Note that `max-width` with `flex` makes the width stretch even when the network name is short.  This PR ensures the PickerNetwork only takes up the space required and ellipsizes once it hits max width.

## Screenshots/Screencaps

SCR-20230502-rbqx
SCR-20230502-rboo

## Manual Testing Steps

1.  Add a variety of networks with different lengths
2. View in full screen
3. View in popup

## Pre-merge author checklist

- [ ] I've clearly explained:
  - [ ] What problem this PR is solving
  - [ ] How this problem was solved
  - [ ] How reviewers can test my changes
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

File: ui/components/ui/box/box.scss [hunk 1]
```diff
@@ -376,7 +376,7 @@ $attributesToApplyExtraProperties: margin;
   @each $breakpoint, $min-width in $screen-sizes-map {
     @each $variant, $color in design-system.$color-map {
       @media screen and (min-width: $min-width) {
-        &--#{$breakpoint}\background-color-#{$variant} {
+        &--#{$breakpoint}\:background-color-#{$variant} {
           background-color: var($color);
         }
       }
```
Comment (georgewrmarshall) [inline ui/components/ui/box/box.scss:382]: Unrelated fixes that I found while using responsive props

File: ui/components/multichain/app-header/app-header.js [hunk 2]
```diff
@@ -128,9 +128,12 @@ export const AppHeader = ({ onClick }) => {
         alignItems={AlignItems.center}
         width={BLOCK_SIZES.FULL}
         backgroundColor={
-          !isUnlocked || popupStatus
+          !isUnlocked
```
Comment (georgewrmarshall) [inline ui/components/multichain/app-header/app-header.js:131]: Can utilize responsive props here so it shows up based on screen size

File: ui/components/multichain/app-header/app-header.js [hunk 3]
```diff
@@ -158,14 +158,22 @@ export const AppHeader = ({ onClick }) => {
...
-                src={currentNetwork?.rpcPrefs?.imageUrl}
-                onClick={networkOpenCallback}
-                display={[DISPLAY.NONE, DISPLAY.FLEX]} // show on desktop hide on popover
-                className="multichain-app-header__contents__network-picker"
-              />
+              {popupStatus ? null : (
+                <Box
+                  display={DISPLAY.FLEX}
+                  className="multichain-app-header__contents__network-container"
+                >
```
Comment (georgewrmarshall) [inline ui/components/multichain/app-header/app-header.js:165]: Instead of using display flex just use a div and the grid wrapper should handle the width so I think you can just do

```suggestion
                <div>
```

File: ui/components/multichain/app-header/app-header.js [hunk 4]
```diff
@@ -158,14 +158,22 @@ export const AppHeader = ({ onClick }) => {
...
+                  <PickerNetwork
+                    margin={2}
+                    label={currentNetwork?.nickname}
+                    src={currentNetwork?.rpcPrefs?.imageUrl}
+                    onClick={networkOpenCallback}
+                    display={[DISPLAY.NONE, DISPLAY.FLEX]} // show on desktop hide on popover
+                    className="multichain-app-header__contents__network-picker"
+                    style={{ maxWidth: '100%' }}
+                  />
+                </Box>
```
Comment (georgewrmarshall) [inline ui/components/multichain/app-header/app-header.js:175]:
```suggestion
                </div>
```

File: ui/components/multichain/app-header/app-header.js [hunk 5]
```diff
@@ -158,14 +158,22 @@ export const AppHeader = ({ onClick }) => {
...
+                  className="multichain-app-header__contents__network-container"
+                >
+                  <PickerNetwork
+                    margin={2}
+                    label={currentNetwork?.nickname}
+                    src={currentNetwork?.rpcPrefs?.imageUrl}
+                    onClick={networkOpenCallback}
+                    display={[DISPLAY.NONE, DISPLAY.FLEX]} // show on desktop hide on popover
+                    className="multichain-app-header__contents__network-picker"
+                    style={{ maxWidth: '100%' }}
```
Comment (georgewrmarshall) [inline ui/components/multichain/app-header/app-header.js:173]:
```suggestion
```

File: ui/components/multichain/app-header/app-header.js [hunk 6]
```diff
@@ -306,12 +314,20 @@ export const AppHeader = ({ onClick }) => {
...
-                label={currentNetwork?.nickname}
-                src={currentNetwork?.rpcPrefs?.imageUrl}
-                onClick={() => dispatch(toggleNetworkMenu())}
-                className="multichain-app-header__contents__network-picker"
-              />
+              {popupStatus ? null : (
+                <Box
+                  display={DISPLAY.FLEX}
+                  className="multichain-app-header__contents__network-container"
+                >
```
Comment (georgewrmarshall) [inline ui/components/multichain/app-header/app-header.js:321]: Grid should handle this
```suggestion
                <div>
```

File: ui/components/multichain/app-header/app-header.js [hunk 7]
```diff
@@ -306,12 +314,20 @@ export const AppHeader = ({ onClick }) => {
...
+                  <PickerNetwork
+                    label={currentNetwork?.nickname}
+                    src={currentNetwork?.rpcPrefs?.imageUrl}
+                    onClick={() => dispatch(toggleNetworkMenu())}
+                    className="multichain-app-header__contents__network-picker"
+                    style={{ maxWidth: '100%' }}
```
Comment (georgewrmarshall) [inline ui/components/multichain/app-header/app-header.js:327]:
```suggestion
```

File: ui/components/multichain/app-header/app-header.js [hunk 8]
```diff
@@ -306,12 +314,20 @@ export const AppHeader = ({ onClick }) => {
...
+                  />
+                </Box>
```
Comment (georgewrmarshall) [inline ui/components/multichain/app-header/app-header.js:329]:
```suggestion
                </div>
```

File: ui/components/multichain/app-header/app-header.scss [hunk 9]
```diff
@@ -29,8 +29,12 @@
       width: $width-screen-lg-min;
     }
 
+    &__network-container {
+      max-width: 250px;
+    }
+
```
Comment (georgewrmarshall) [inline ui/components/multichain/app-header/app-header.scss:35]: Shouldn't need this

```suggestion
```

File: ui/components/multichain/app-header/app-header.scss [hunk 10]
```diff
@@ -29,8 +29,12 @@
...
     &__network-picker {
-      max-width: 200px;
+      max-width: 100%;
```
Comment (georgewrmarshall) [inline ui/components/multichain/app-header/app-header.scss:37]: Return this to whatever pixel value you want
```suggestion
      max-width: 250px;
```

**Review by georgewrmarshall — COMMENTED** [review]
Tried some things and pushed it included some other changes so have just reverted and added comments

**Review by georgewrmarshall — DISMISSED** [review]
LGTM. Checked this on the storybook build of this PR so it doesn't register popupStatus. I wonder if we can change it to use `useMediaQuery` hook or something so it's screensize based and works in storybook

Screenshot 2023-05-05 at 11 40 08 AM

**Review by vthomas13 — DISMISSED** [review]

**Review by georgewrmarshall — APPROVED** [review]
LGTM

**Comment by Ehsan129** [comment]
Thank you MetaMask...

1299820

در تاریخ جمعه ۵ مه ۲۰۲۳،‏ ۱۸:۴۱ MetaMask Bot ***@***.***>
نوشت:

**Review by vthomas13 — APPROVED** [review]

**Review by vthomas13 — APPROVED** [review]
